### DISCUSSION TITLE
Tooltip with delay

### DISCUSSION THREAD

**Discussion by knorle** [hecrj/iced#1291] [Q&A]
I would like to show a tooltip only once the mouse pointer has been hovering over a button for a certain amount of time.

How could one achieve this?

**Comment by lufte** [comment]
I will need to do this soon in the app I'm developing. I don't have a clear answer yet, but what I was thinking is something like this:

1. Have a tooltip component of some sort already positioned over the button, but don't display it.
2. Have a field in your state that will hold the last time your button was hovered, let's call it `last_hover_timestamp`.
3. When your button is hovered (I don't know if Iced provides a way to capture this kind of event yet) fire a message with the current timestamp.
4. In your update function, when you handle this message, you will set `last_hover_timestamp` to the timestamp stored inside the message.
5. In your view function, you will set your tooltip component to visible every time that `current_timestamp() < last_hover_timestamp + delay` where `delay` is the amount of time you wish your tooltip to stay visible.

I hope this helps you.

**Reply by knorle** [reply]
Thanks for your input. I ended up doing it more or less how you suggested.

The issue I encountered was that the UI does not update, when there is no event, e.g. when the mouse cursor hovers the button but is not moving. I crudely solved this by firing a message on mouse enter, that starts a command which in turn fires a message after a certain delay to update the UI.

**Reply by lufte** [reply]
Exactly! It turned out my original answer would not work as the `view` function does not execute until there is an incoming message. I did the exact same thing in my app, which I think is fine as there is no other way to do it yet. Having said that, there is some progress to add a different approach #1284
